apps/spruce/src/pages/image/tabs/EventLogTab/EventLogTab.test.tsx

```diff
@@ -0,0 +1,395 @@
+      ImageEventEntryAction.Updated,
+    );
+    const row1 = within(card0).getAllByDataCy("image-event-log-table-row")[1];
+    expect(within(row1).getAllByRole("cell")[4]).toHaveTextContent(
+      ImageEventEntryAction.Added,
+    );
+    const row2 = within(card0).getAllByDataCy("image-event-log-table-row")[2];
+    expect(within(row2).getAllByRole("cell")[4]).toHaveTextContent(
+      ImageEventEntryAction.Deleted,
+    );
+  });
+
+  it("supports filtering for action field", async () => {
+    const { Component } = RenderFakeToastContext(<EventLogTab />);
+    render(<Component />, { wrapper });
+    await waitFor(() => {
+      expect(screen.queryAllByDataCy("image-event-log-card")).toHaveLength(5);
+    });
+    const card0 = screen.getAllByDataCy("image-event-log-card")[0];
+    within(card0).getByDataCy("image-event-log-action-filter").click();
+    const treeSelectOptions = await screen.findByDataCy("tree-select-options");
+    within(treeSelectOptions).getByText("UPDATED").click();
+    await waitFor(() => {
+      const rows = within(card0).getAllByDataCy("image-event-log-table-row");
+      expect(rows).toHaveLength(1);
+    });
+    within(treeSelectOptions).getByText("UPDATED").click();
+    await waitFor(() => {
+      const rows = within(card0).getAllByDataCy("image-event-log-table-row");
+      expect(rows).toHaveLength(3);
+    });
+    within(treeSelectOptions).getByText("ADDED").click();
+    await waitFor(() => {
+      const rows = within(card0).getAllByDataCy("image-event-log-table-row");
+      expect(rows).toHaveLength(1);
+    });
+    within(treeSelectOptions).getByText("ADDED").click();
+    await waitFor(() => {
+      const rows = within(card0).getAllByDataCy("image-event-log-table-row");
+      expect(rows).toHaveLength(3);
+    });
+    within(treeSelectOptions).getByText("DELETED").click();
+    await waitFor(() => {
+      const rows = within(card0).getAllByDataCy("image-event-log-table-row");
+      expect(rows).toHaveLength(1);
+    });
+    within(treeSelectOptions).getByText("DELETED").click();
+    await waitFor(() => {
+      const rows = within(card0).getAllByDataCy("image-event-log-table-row");
+      expect(rows).toHaveLength(3);
+    });
+  });
+});
+
+const imageEventsMock: ApolloMock<ImageEventsQuery, ImageEventsQueryVariables> =
+  {
+    request: {
+      query: IMAGE_EVENTS,
+      variables: {
+        imageId: "ubuntu2204",
+        limit: 5,
+        page: 0,
+      },
+    },
+    result: {
+      data: {
+        image: {
+          __typename: "Image",
+          id: "ubuntu2204",
+          events: {
+            count: 5,
+            eventLogEntries: [
+              {
+                amiAfter: "ami-03bfb241d1718c8a2",
+                amiBefore: "ami-03e245926032896f9",
+                entries: [
+                  {
+                    // @ts-expect-error: Actual provided input from API does not match ImageEventType.Package type
+                    type: "Packages",
```

Comment:
Ahh I see what you're saying now... To be honest, it kind of seems like a GraphQL error since this is what we're expecting the schema to return and maybe we need to update some logic in the backend. Let me check on this